File: GestaoLivrosApi/GestaoLivrosApi/Models/Authors/CreateAuthorDTO.cs
using System;
using FluentValidation;
using Microsoft.AspNetCore.Http.HttpResults;

namespace GestaoLivrosApi.Models.Authors
{
    public class CreateAuthorDTO
    {
        public string Name { get; set; } = null!;

        public string Country { get; set; } = null!;

        //public ICollection<Book> Books { get; set; }

        public string? Image { get; set; }

        public Author ToEntity()
        {
            var author = new Author
            {
                Name = this.Name,
                Country = this.Country,
                Image = this.Image
            };
            return author;
        }

        public class CreateAuthorDTOValidator : AbstractValidator<CreateAuthorDTO>
        {
            public CreateAuthorDTOValidator()
            {
                RuleFor(a => a.Name).NotNull().WithMessage("Insira o Nome do autor");
                RuleFor(a => a.Country).NotNull().WithMessage("Insira o País do autor");
            }
        }

    }
}

File: frontend/src/models/Authors/CreateAuthorDTO.ts
import Joi from "joi";

export class CreateAuthorDTO{
    name:string = "";
    country:string = "";
    image:string = "";
    books:[] =[];
}

export const CreateAuthorDTOSchema = Joi.object({
    name: Joi.string().messages({"string.empty":"Nome do autor deve ser preenchido"}),
    country: Joi.string().messages({"string.empty":"País do autor deve ser preenchido"})
})